new-log-viewer/src/typings/decoders.ts

```diff
@@ -42,33 +49,49 @@ interface Decoder {
     getEstimatedNumEvents(): number;
 
     /**
-     * When applicable, deserializes log events in the range `[beginIdx, endIdx)`.
+     * @return Indices of the filtered events.
+     */
+    getFilteredLogEventMap(): FilteredLogEventMap;
+
+    /**
+     * Sets the log level filter for the decoder.
+     *
+     * @param logLevelFilter
+     * @return Whether the filter was successfully set.
+     */
+    setLogLevelFilter(logLevelFilter: LogLevelFilter): boolean
+
+    /**
+     * Deserializes all log events in the file.
      *
-     * @param beginIdx
-     * @param endIdx End index. To deserialize to the end of the file, use `LOG_EVENT_FILE_END_IDX`.
      * @return Count of the successfully deserialized ("valid") log events and count of any
-     * un-deserializable ("invalid") log events within the range; or null if any log event in the
-     * range doesn't exist (e.g., the range exceeds the number of log events in the file).
+     * un-deserializable ("invalid") log events within the range;
      */
-    buildIdx(beginIdx: number, endIdx: number): Nullable<LogEventCount>;
+    build(): LogEventCount;
 
     /**
-     * Sets options for the decoder.
+     * Sets formatting options. Decoders support changing formatting without rebuilding
+     * existing log events.
      *
      * @param options
      * @return Whether the options were successfully set.
      */
-    setDecoderOptions(options: DecoderOptionsType): boolean;
+    setFormatterOptions(options: DecoderOptionsType): boolean;
 
     /**
-     * Decodes the log events in the range `[beginIdx, endIdx)`.
+     * Decode log events. The range boundaries `[BeginIdx, EndIdx)` can refer to unfiltered log
+     * event indices or filtered log event indices based on the flag `useFilter`.
      *
      * @param beginIdx
      * @param endIdx
+     * @param useFilter Whether to decode from the filtered or unfiltered log events array.
      * @return The decoded log events on success or null if any log event in the range doesn't exist
      * (e.g., the range exceeds the number of log events in the file).
      */
-    decode(beginIdx: number, endIdx: number): Nullable<DecodeResultType[]>;
+    decodeRange(beginIdx: number,
```

Comment:
```suggestion
    decodeRange(
        beginIdx: number,
```

Until we improve our automated formatting, let's try to follow this style.